binding.cc

```diff
@@ -258,18 +253,34 @@ namespace zmq {
   void
   Context::Close() {
     if (context_ != NULL) {
-      if (zmq_term(context_) < 0) throw std::runtime_error(ErrorMessage());
+      int rc;
+
+      while (true) {
+        #if ZMQ_VERSION < 30200
+          rc = zmq_term(context_);
+        #else
+          rc = zmq_ctx_destroy(context_);
+        #endif
+        if (rc < 0) {
```

Comment:
style nit: mixing macro conditions with normal condition declarations over the same control value